src/components/application-events/views.test.tsx

```diff
@@ -0,0 +1,52 @@
+import cheerio from 'cheerio';
+import { shallow } from 'enzyme';
+import React from 'react';
+
+import { IApplication, IAuditEvent } from '../../lib/cf/types';
+import { ApplicationEventsPage } from './views';
+
+describe(ApplicationEventsPage, () => {
+  const application = {metadata: {guid: 'APPLICATION_GUID'}, entity: {name: 'test-app'}} as unknown as IApplication;
```

Comment:
I'm curious about this bit of TypeScript. Does `{ … } as unknown as Type` typecheck where `{ … } as Type` would not?

Reply:
This is to do with the linter more than a compiler.

Imagine a scenario, where your interface requires 5 parameters, but you are certain that the function using the interface is only using 1 parameter.

You don't necessarily want to compose the entire object (considering CF API), just to check if the `name` parameter is displayed correctly.

Linter will tell complain, that your custom object, is missing the remaining 4 parameters.

If you cast it to unknown first, and then to the type, it gets to the state where it forgets that the object is missing pieces inside...

It's a terrible hack... Should not be used in production code...

Reply:
Oh I see! Thanks Rafal. 😄 I think that's OK in ordinary production code, but maybe not in scary places.